app/src/hid.c

```diff
@@ -131,10 +134,116 @@ int zmk_hid_consumer_release(zmk_key_t code) {
 
 void zmk_hid_consumer_clear() { memset(&consumer_report.body, 0, sizeof(consumer_report.body)); }
 
+// Keep track of how often a button was pressed.
+// Only release the button if the count is 0.
+static int explicit_button_counts[16] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+static zmk_mod_flags_t explicit_buttons = 0;
+static int16_t curr_x = 0;
+static int16_t curr_y = 0;
+static int8_t curr_hor = 0;
+static int8_t curr_vert = 0;
+
+#define SET_MOUSE_BUTTONS(butts)                                                                   \
```

Comment:
`butts` -> `buttons`

Reply:
Due to the way macros work, that would also impact the `buttons` in `mouse_report.body.buttons`. This was a bug we faced during development.

Reply:
ah right, make those uppercase `BUTTONS`

Reply:
btns would be a better shorthand than butts

Reply:
Implemented @mcrosson's suggestion.